<a href="https://colab.research.google.com/github/yshzjq/Step-1.-Python-SQL-/blob/main/%EC%B1%95%ED%84%B03_Pandas_Matplotlib%C2%B7Seaborn/3_4.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [4]:
# === Pandas로 결츨치 평균 계산 (3줄) ===
import pandas as pd
df = pd.read_csv('seoul-rainfall-data.csv', encoding='cp949')
print(df['강수량(mm)'].mean())

print(df)


8.196250000000001
             날짜   지점  강수량(mm)
0    2024-01-01  108      0.0
1    2024-01-02  108      0.0
2    2024-01-03  108      0.6
3    2024-01-04  108      NaN
4    2024-01-05  108      0.0
..          ...  ...      ...
361  2024-12-27  108      0.0
362  2024-12-28  108      NaN
363  2024-12-29  108      NaN
364  2024-12-30  108      0.0
365  2024-12-31  108      0.0

[366 rows x 3 columns]


In [5]:
df.head(3)
df.head(10)

,날짜,지점,강수량(mm)
0,2024-01-01,108,0.0
1,2024-01-02,108,0.0
2,2024-01-03,108,0.6
3,2024-01-04,108,NaN
4,2024-01-05,108,0.0
5,2024-01-06,108,2.4
6,2024-01-07,108,NaN
7,2024-01-08,108,NaN
8,2024-01-09,108,1.6
9,2024-01-10,108,NaN


In [6]:
df.shape

(366, 3)

In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 366 entries, 0 to 365
Data columns (total 3 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   날짜       366 non-null    object 
 1   지점       366 non-null    int64  
 2   강수량(mm)  160 non-null    float64
dtypes: float64(1), int64(1), object(1)
memory usage: 8.7+ KB


In [8]:
df.describe()

,지점,강수량(mm)
count,366.0,160.000000
mean,108.0,8.196250
std,0.0,17.687642
min,108.0,0.000000
25%,108.0,0.000000
50%,108.0,1.100000
75%,108.0,7.825000
max,108.0,128.800000


In [9]:
import pandas as pd

# 세 가지 데이터
A = pd.Series([5, 5, 5, 5, 5])      # 전부 같음
B = pd.Series([3, 4, 5, 6, 7])      # 조금 퍼짐
C = pd.Series([0, 0, 5, 10, 10])    # 많이 퍼짐

print(f"A: 평균={A.mean()}, 표준편차={A.std():.2f}")
print(f"B: 평균={B.mean()}, 표준편차={B.std():.2f}")
print(f"C: 평균={C.mean()}, 표준편차={C.std():.2f}")

A: 평균=5.0, 표준편차=0.00
B: 평균=5.0, 표준편차=1.58
C: 평균=5.0, 표준편차=5.00


In [10]:
df.columns # 컬럼명 확인

Index(['날짜', '지점', '강수량(mm)'], dtype='object')

In [26]:
import pandas as pd

df = pd.read_csv('Sample-Superstore.csv', encoding='latin1')

df.head(5)


,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,...,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit
0,1,CA-2016-152156,11/8/2016,11/11/2016,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,...,42420,South,FUR-BO-10001798,Furniture,Bookcases,Bush Somerset Collection Bookcase,261.9600,2,0.00,41.9136
1,2,CA-2016-152156,11/8/2016,11/11/2016,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,...,42420,South,FUR-CH-10000454,Furniture,Chairs,"Hon Deluxe Fabric Upholstered Stacking Chairs,...",731.9400,3,0.00,219.5820
2,3,CA-2016-138688,6/12/2016,6/16/2016,Second Class,DV-13045,Darrin Van Huff,Corporate,United States,Los Angeles,...,90036,West,OFF-LA-10000240,Office Supplies,Labels,Self-Adhesive Address Labels for Typewriters b...,14.6200,2,0.00,6.8714
3,4,US-2015-108966,10/11/2015,10/18/2015,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,...,33311,South,FUR-TA-10000577,Furniture,Tables,Bretford CR4500 Series Slim Rectangular Table,957.5775,5,0.45,-383.0310
4,5,US-2015-108966,10/11/2015,10/18/2015,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,...,33311,South,OFF-ST-10000760,Office Supplies,Storage,Eldon Fold 'N Roll Cart System,22.3680,2,0.20,2.5164


In [27]:
df.shape

(9994, 21)

In [28]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9994 entries, 0 to 9993
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Row ID         9994 non-null   int64  
 1   Order ID       9994 non-null   object 
 2   Order Date     9994 non-null   object 
 3   Ship Date      9994 non-null   object 
 4   Ship Mode      9994 non-null   object 
 5   Customer ID    9994 non-null   object 
 6   Customer Name  9994 non-null   object 
 7   Segment        9994 non-null   object 
 8   Country        9994 non-null   object 
 9   City           9994 non-null   object 
 10  State          9994 non-null   object 
 11  Postal Code    9994 non-null   int64  
 12  Region         9994 non-null   object 
 13  Product ID     9994 non-null   object 
 14  Category       9994 non-null   object 
 15  Sub-Category   9994 non-null   object 
 16  Product Name   9994 non-null   object 
 17  Sales          9994 non-null   float64
 18  Quantity

In [29]:
import pandas as pd

df = pd.read_csv('Sample-Superstore.csv', encoding='latin-1')
print(df.describe())


            Row ID   Postal Code         Sales     Quantity     Discount  \
count  9994.000000   9994.000000   9994.000000  9994.000000  9994.000000   
mean   4997.500000  55190.379428    229.858001     3.789574     0.156203   
std    2885.163629  32063.693350    623.245101     2.225110     0.206452   
min       1.000000   1040.000000      0.444000     1.000000     0.000000   
25%    2499.250000  23223.000000     17.280000     2.000000     0.000000   
50%    4997.500000  56430.500000     54.490000     3.000000     0.200000   
75%    7495.750000  90008.000000    209.940000     5.000000     0.200000   
max    9994.000000  99301.000000  22638.480000    14.000000     0.800000   

            Profit  
count  9994.000000  
mean     28.656896  
std     234.260108  
min   -6599.978000  
25%       1.728750  
50%       8.666500  
75%      29.364000  
max    8399.976000  


평균 매출 229

최대 매출 22638

평균 이익 28

이익 최소값 -6599

-> 음수로 표기되어 원인 분석 필요






In [34]:
import pandas as pd
df = pd.read_csv('Sample-Superstore.csv', encoding='latin-1')

df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9994 entries, 0 to 9993
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Row ID         9994 non-null   int64  
 1   Order ID       9994 non-null   object 
 2   Order Date     9994 non-null   object 
 3   Ship Date      9994 non-null   object 
 4   Ship Mode      9994 non-null   object 
 5   Customer ID    9994 non-null   object 
 6   Customer Name  9994 non-null   object 
 7   Segment        9994 non-null   object 
 8   Country        9994 non-null   object 
 9   City           9994 non-null   object 
 10  State          9994 non-null   object 
 11  Postal Code    9994 non-null   int64  
 12  Region         9994 non-null   object 
 13  Product ID     9994 non-null   object 
 14  Category       9994 non-null   object 
 15  Sub-Category   9994 non-null   object 
 16  Product Name   9994 non-null   object 
 17  Sales          9994 non-null   float64
 18  Quantity

In [35]:
print(df['Category'].unique())

['Furniture' 'Office Supplies' 'Technology']


In [36]:
print(df['Region'].unique())

['South' 'West' 'Central' 'East']


In [38]:
print(df['Segment'].unique())
print(df['Segment'].nunique())

['Consumer' 'Corporate' 'Home Office']
3
